src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -976,4 +978,34 @@ public static void onAdvancementProgressedEvent(Player player, Advancement progr
     {
         MinecraftForge.EVENT_BUS.post(new AdvancementProgressEvent(player, progressed, advancementProgress, criterion, progressType));
     }
+
+    /**
+     * Fires {@link GetEnchantmentLevelEvent} and for a single enchantment, returning the (possibly event-modified) level.
+     * 
+     * @param level The original level of the enchantment as provided by the Item.
+     * @param stack The stack being queried against.
+     * @param ench  The enchantment being queried for.
+     * @return The new level of the enchantment.
+     */
+    public static int getEnchantmentLevelSpecific(int level, ItemStack stack, Enchantment ench) {
+        Map<Enchantment, Integer> map = new HashMap<>();
```

Comment:
Is the map supposed to be ordered?

Reply:
Logical uses of enchantments should not be relying on the order of enchantments in the map. This event is not modifying enchantments in NBT afterall, just in gameplay logic.

Reply:
No, ordering is not relevant for an enchantment map.